src/main/java/dev/jbang/source/ProjectBuilder.java

```diff
@@ -381,6 +409,21 @@ private Project importJarMetadata(Project prj, boolean importModuleName) {
 						// buildJdk = JavaUtil.parseJavaVersion(ver);
 						prj.setJavaVersion(JavaUtil.parseJavaVersion(ver) + "+");
 					}
+
+					// we pass exports/opens into the project...
+					// TODO: this does mean we can't separate from user specified options and jar
+					// origined ones
+					// but not sure if needed?
+					// https://openjdk.org/jeps/261#Breaking-encapsulation
+					String exports = attrs.getValue("Add-Exports");
+					if (exports != null) {
+						prj.getManifestAttributes().put("Add-Exports", exports);
+					}
+					String opens = attrs.getValue("Add-Opens");
+					if (opens != null) {
+						prj.getManifestAttributes().put("Add-Opens", exports);
```

Comment:
Indeed. As an alternative , I don't know if it would be useful/better/easier to just blindly copy all options. But that's something we could change at any time in the future.